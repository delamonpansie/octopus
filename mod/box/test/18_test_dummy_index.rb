#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
rows_per_wal=100

object_space[0].index[0].key_field[0].type = "NUM"

object_space[0].index[1].type = "TREE"
object_space[0].index[1].unique = 0
object_space[0].index[1].key_field[0].fieldno = 0
object_space[0].index[1].key_field[0].type = "NUM"

EOD
  end
end

env = Env.new

env.start
env.connect_eval do
  ping
  insert [1,2]
  insert [2,3]

  env.snapshot
end

env.restart
env.connect_eval do
  ping
  insert [1,2]
  insert [2,3]
end

env.restart
env.connect_eval do
  ping
  insert [1,2]
  insert [2,3]
end


