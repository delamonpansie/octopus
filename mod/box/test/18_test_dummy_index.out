# box.ping()
:pong

# box.insert([1, 2])
1

# box.insert([2, 3])
1

# box.ping()
:pong

# box.insert([1, 2])
2

# box.insert([2, 3])
2

# box.ping()
:pong

# box.insert([1, 2])
2

# box.insert([2, 3])
2

